hazelcast-client/src/test/java/com/hazelcast/client/ClientConnectionTest.java

```diff
@@ -0,0 +1,50 @@
+package com.hazelcast.client;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+/**
+ * User: sancar
+ * Date: 15/01/15
+ * Time: 13:29
+ */
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class ClientConnectionTest extends HazelcastTestSupport {
+
+    @After
+    @Before
+    public void cleanup() {
+        HazelcastClient.shutdownAll();
+        Hazelcast.shutdownAll();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testWithIllegalAddress() {
+        String illegalAddress = randomString();
+
+        Hazelcast.newHazelcastInstance();
+        ClientConfig config = new ClientConfig();
+        config.getNetworkConfig().addAddress(illegalAddress);
+        HazelcastClient.newHazelcastClient(config);
+    }
+
+    @Test
+    public void testWithLegalAndIllegalAddressTogether() {
+        String illegalAddress = randomString();
+
+        Hazelcast.newHazelcastInstance();
+        ClientConfig config = new ClientConfig();
+        config.getNetworkConfig().addAddress(illegalAddress).addAddress("localhost");
+        HazelcastClient.newHazelcastClient(config);
```

Comment:
shouldn't you check the client has actually connected to the member running locally?